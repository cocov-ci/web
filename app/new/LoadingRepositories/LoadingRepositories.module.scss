.alert {
  margin-top: 30px;
}
