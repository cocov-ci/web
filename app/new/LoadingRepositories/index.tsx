'use client'

import Alert from 'app/common/Alert'

import styles from './LoadingRepositories.module.scss'

const LoadingRepositories = () => {
  return (
    <Alert
      className={styles.alert}
      description="Wait a moment while Cocov loads the repository list from GitHub. This shouldn’t take long."
      loading
      title="Loading Repositories…"
    />
  )
}

export default LoadingRepositories
