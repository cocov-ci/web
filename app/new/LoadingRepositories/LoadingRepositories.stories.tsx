import type { Meta, StoryObj } from '@storybook/react'

import LoadingRepositories from './index'

const meta: Meta<typeof LoadingRepositories> = {
  title: 'Repositories/New/LoadingRepositories',
  component: LoadingRepositories,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof LoadingRepositories>

export const Default: Story = {
  args: {},
}
