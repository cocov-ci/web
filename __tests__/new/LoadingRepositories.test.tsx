import { render, screen } from '@testing-library/react'

import LoadingRepositories from 'app/new/LoadingRepositories'

import '@testing-library/jest-dom'

describe('new/LoadingRepositories', () => {
  it('renders LoadingRepositories', () => {
    const { container } = render(<LoadingRepositories />)

    expect(screen.getByText('Loading Repositoriesâ€¦')).toBeVisible()
    expect(container.querySelector('img')).toBeVisible()
  })

  it('renders LoadingRepositories snapshots', () => {
    const { container } = render(<LoadingRepositories />)

    expect(container).toMatchSnapshot()
  })
})
