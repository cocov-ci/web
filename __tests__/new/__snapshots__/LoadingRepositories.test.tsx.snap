// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new/LoadingRepositories renders LoadingRepositories snapshots 1`] = `
<div>
  <div
    class="alert alert"
  >
    <div
      class="box"
    >
      <div
        class="loading center loading"
      >
        <img
          alt="A spinner indicating an ongoing operation"
          class="spinner"
          data-nimg="1"
          decoding="async"
          height="30"
          loading="lazy"
          src="/icons/spinner.svg"
          srcset="/icons/spinner.svg 1x, /icons/spinner.svg 2x"
          style="color: transparent;"
          width="30"
        />
      </div>
      <div
        class="content"
      >
        <h2>
          Loading Repositories…
        </h2>
        <h3>
          Wait a moment while Cocov loads the repository list from GitHub. This shouldn’t take long.
        </h3>
      </div>
    </div>
  </div>
</div>
`;
